test/src/main/java/org/apache/accumulo/test/shell/ShellServerIT.java

```diff
@@ -1321,51 +1315,61 @@ public void testGetGoalCommand() throws IOException, TableNotFoundException {
 
   // Verify that when splits are added after table creation, hosting goals are set properly
   @Test
-  public void testGetGoalCommand_DelayedSplits() throws IOException {
+  public void testGetGoalCommand_DelayedSplits() throws Exception {
+
+    for (int i = 0; i < 40; i++) {
+      ts.exec("createtable tab" + i, true);
+    }
 
     final String[] tableName = getUniqueNames(2);
 
     ts.exec("createtable " + tableName[0], true);
+
+    String tableId = getTableId(tableName[0]);
+
     String result = ts.exec("getgoal");
+
     assertTrue(result.contains("TABLE: " + tableName[0]));
     assertTrue(result.contains("TABLET ID    HOSTING GOAL"));
-    assertTrue(result.contains("1<<          ONDEMAND"));
+    assertTrue(result.matches("(?s).*" + tableId + "<<\\s+ONDEMAND.*"));
```

Comment:
Are the lines in the output coming in an unpredictable order? Is there a reason why we're doing pattern matching anywhere in the multi-line string result for the lines we expect, instead of just iterating over the lines and checking to verify each line contains the next thing we expect?

Something like:

```java
    Iterator<String> iter = splitLines(result);
    line = iter.next(); assertTrue(line.contains(firstLineCondition));
    line = iter.next(); assertTrue(line.contains(nextLineCondition));
    line = iter.next(); assertTrue(line.contains(nextLineCondition));
    // ...
    assertFalse(iter.hasNext());
```

This seems simpler to read and maintain than a bunch of regexes that don't verify that the output is appearing in the expected order.

Reply:
The reason I did pattern matching was because the number of spaces in the output would change when the length of the table id changed.  So for a one character table id would see more spaces in the output than if there was a two digit table id.

As for checking the order, that could be a follow on issue.  It would make the test more strict.  The test was not checking the order before these changes.